docs/source/architecture/injecting_batches_block_validation_rules.rst

```diff
@@ -0,0 +1,150 @@
+The BatchInjector class supports injecting transactions into blocks:
+
+.. code-block:: python
+
+  interface BatchInjector:
+    // Called when a new block is created and before any batches are added. A list of
+    // batches to insert at the beginning of the block must be returned. A StateView
+    // is provided for inspecting state as of the previous block.
+    block_start(string previous_block_id) -> list<Batch>
+
+    // Called before inserting the incoming batch into the pending queue for the
+    // given block. A list of batches to insert before this batch must be returned.
+    before_batch(string previous_block_id, Batch batch) -> list<Batch>
+
+    // Called after inserting the incoming batch into the pending queue for the
+    // given block. A list of batches to insert after this batch must be returned.
+    after_batch(string previous_block_id, Batch batch) -> list<Batch>
+
+    // Called just before finalizing and completing a Block. An ordered list of batches
+    // that will be committed in the block is passed in. A list of batches to insert at
+    // the end of the block must be returned.
+    block_end(string previous_block_id, list<Batch> batches) -> list<Batch>
+
+The BatchInjector interface that will be implemented by concrete classes which may
+hold references to additional data internally
+
+New classes will be created that implement this interface.
+The actual BatchInjectors that are loaded will be configured dynamically using
+an on-chain setting described below. The Journal will call each of the methods
+at the appropriate times for all loaded implementations.
+
+On-Chain Configuration
+======================
+Similar to how the consensus module loaded by the validator is configured using
+an on-chain setting, the set of BatchInjectors to load is configured with an
+on-chain setting. The setting key is sawtooth.validator.batch_injectors and
+stores a comma-separated list of batch injectors to load. This list is parsed
+by the validator at the beginning of block publishing for each block and the
+appropriate python modules will be loaded using importlib. The BatchInjector
+defined in that module is then instantiated. If the module cannot be loaded or
+an instance of the class cannot be created, block publishing will continue
+without the BatchInjector.
+
+This setting is controlled using the existing Settings Transaction Family. Care
+will need to be taken when updating this setting, as an incorrect value may
+cause transaction families to behave incorrectly.
+
+On-Chain Validation Rules
+=========================
+In order to support the pluggable validation logic requirement, an on-chain
+setting will hold a set of validation rules that are enforced for each block.
+The setting key where the validation rules are stored is
+sawtooth.validator.block_validation_rules and it will contain a string
+representation of the additional rules to enforce. The on-chain rules will be
+enforced by the block validator.
```

Comment:
Change "The setting key..enforced by the block validator." to "On-chain validation rules will be stored as a string in the setting key sawtooth.validator.block_validation_rules. Rules will be enforce by the block validator."